internal/cri/server/container_create.go

```diff
@@ -247,8 +247,15 @@ func (c *criService) CreateContainer(ctx context.Context, r *runtime.CreateConta
 			sandboxConfig.GetLogDirectory(), config.GetLogPath())
 	}
 
-	containerIO, err := cio.NewContainerIO(id,
-		cio.WithNewFIFOs(volatileContainerRootDir, config.GetTty(), config.GetStdin()))
+	var containerIO *cio.ContainerIO
+	switch ociRuntime.IOType {
+	case "streaming":
+		containerIO, err = cio.NewContainerIO(id,
```

Comment:
do we miss this part in recover stage during bootstrap of containerd?

Reply:
I think we don't need to because the streaming url is alread set into the Stdin, Stdout and Stderr of a task, and we will recover the io transfer by opening connection there if the Stdin/Stdout/Stderr is a streaming format. https://github.com/containerd/containerd/blob/main/internal/cri/server/restart.go#L302